### PR TITLE
Citations generation: Config and React app

### PR DESCRIPTION
**Pull request by ines-cruz**
* closes https://github.com/inveniosoftware/invenio-app-rdm/issues/867
* needs https://github.com/inveniosoftware/invenio-rdm-records/pull/648

https://user-images.githubusercontent.com/15194802/121540329-b0d07c80-ca06-11eb-8503-b94f26454d2e.mp4

### CODE REVIEW

**Comment by ines-cruz** [comment]
citations1

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js
Comment (fenekku) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:433]: Per https://github.com/inveniosoftware/react-invenio-deposit/pull/299#pullrequestreview-675728912 , the citation is generated on the landing page. It shouldn't be on the deposit page.

**Review by fenekku — COMMENTED** [review]
Like https://github.com/inveniosoftware/react-invenio-deposit/pull/299#pullrequestreview-675728912 , some re-orientation is needed.

**Comment by jrcastro2** [comment]
Once it's approved I will squash the commits

File: invenio_app_rdm/config.py
Comment (ppanero) [inline invenio_app_rdm/config.py:518]: what do the `TODO` stand for? which value should be corrected?

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js
Comment (ppanero) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js:23]: This should be removed right?

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js
Comment (ppanero) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js:75]: `e` is a bit ambiguous, maybe a better naming for this variable?

**Review by ppanero — COMMENTED** [review]
Some minor comments. LGTM, but it's blocked until the `CitationStringSerializer` is merged in rdm-records. There are no new screenshots, I guess the old ones are still working in the same manner, right?

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js:51]:
```suggestion

```
can be moved outside the if, only once.

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js:90]: can you add the props types and defaults please?

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme.less
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme.less:420]: since it is only one in the page, maybe we could use an `id` to group classes for the same style. This is also faster as CSS selector than CSS classes `.`. e.g:

```
#record-citation: {
...

    .citation-style-selector {...}

}

**Review by ntarocco — COMMENTED** [review]
🚀 💯

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js
Comment (jrcastro2) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js]: Changed the component to be a class component and added the cancellable pattern, thanks for the good advice!

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js:1]: It might be much more verbose, but IMO is much more readable and understandable :)

**Review by ntarocco — APPROVED** [review]
Looks very good to me! Good job! 💯

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/RecordCitationField.js:53]:
```suggestion
          Accept: "text/x-bibliography",
```

**Pull request merged**
